internal/pkg/gateway/registry.go

```diff
@@ -264,63 +266,54 @@ func (reg *registry) orderers(channel string) ([]*orderer, error) {
 	if err != nil {
 		return nil, err
 	}
-	var orderers []*orderer
 
-	// Get the config
-	config, err := reg.discovery.Config(channel)
-	if err != nil {
-		return nil, err
-	}
-
-	reg.configLock.RLock()
-	defer reg.configLock.RUnlock()
-
-	for _, eps := range config.GetOrderers() {
-		for _, ep := range eps.Endpoint {
-			url := fmt.Sprintf("%s:%d", ep.Host, ep.Port)
-			if ord, ok := reg.broadcastClients[url]; ok {
-				orderers = append(orderers, ord)
-			}
-		}
+	var orderers []*orderer
+	for _, url := range reg.channelOrderers[channel] {
```

Comment:
Can't this map be accessed concurrently while **registerChannel()** is invoked at the same time? You're only locking in **registerChannel** but not here.

Reply:
Yes, I'd accidentally removed the read lock in this commit.
I've now modified the logic so that it doesn't access/change anything other than sync.Maps